crates/datasources/src/native/access.rs

```diff
@@ -112,6 +112,27 @@ impl NativeTableStorage {
         Ok(NativeTable::new(table))
     }
 
+    pub async fn delete_table(&self, table: &TableEntry) -> Result<()> {
+        let prefix = format!("databases/{}/tables/{}", self.db_id, table.meta.id);
+        let path = match &self.conf {
+            StorageConfig::Gcs { bucket, .. } => format!("gs://{}/{}", bucket, prefix).into(),
+            StorageConfig::Local { path } => {
+                let path =
+                    fs::canonicalize(path)
+                        .await
+                        .map_err(|e| NativeError::CanonicalizePath {
+                            path: path.clone(),
+                            e,
+                        })?;
+                let path = path.join(prefix);
+                format!("{:?}", path.to_str()).into()
+            }
+            StorageConfig::Memory => format!("memory://{}", prefix).into(),
+        };
+        self.store.delete(&path).await?;
+        Ok(())
+    }
```

Comment:
Let's just extract out the path creation from `create_delta_store_for_table` and reuse that logic here. This is trying to convert a path to string to an object store path, which leads to an incorrect conversion.

Reply:
makes sense. I will update it

Reply:
delete takes a relative path, which was the actual issue